File: 백준/Silver/10828. 스택/스택.py
from collections import deque
import sys
input = sys.stdin.readline

n = int(input())

stack = deque()

# push pop empty size top
for _ in range(n):
    ins = input()
    ins = ins.split()

    if ins[0] == 'push':
        stack.append(int(ins[1]))
    elif ins[0] == 'pop':
        if len(stack) == 0:
            print(-1)
        else:
            print(stack.pop())
    elif ins[0] == 'size':
        print(len(stack))
    elif ins[0] == 'top':
        if len(stack) == 0:
            print(-1)
        else:
            print(stack[len(stack)-1])
    elif ins[0] == 'empty':
        if len(stack) == 0:
            print(1)
        else:
            print(0)

File: 백준/Silver/10828. 스택/스택.js
const input = require('fs').readFileSync(process.platform == 'win32' ? "./input.txt" : '/dev/stdin').toString().trim().split("\n")

const solution = () => {
    const n = Number(input[0]);
    let stack = [];
    let answer = [];

    for (let i = 1; i < n+1; i++) {
        const command = input[i].split(" ");
        

        switch (command[0].trim()) {
            case "push": stack.push(Number(command[1])); break;
            case "pop": stack.length != 0 ? answer.push(stack.pop()) : answer.push(-1); break;
            case "size": answer.push(stack.length); break;
            case "empty": stack.length == 0 ? answer.push(1) : answer.push(0); break;
            case "top": stack.length == 0 ? answer.push(-1) : answer.push(stack[stack.length - 1]); break;
        }
    }

    console.log(answer.join("\n"));
}

solution();